lib/src/pubspec.dart

```diff
@@ -116,9 +116,35 @@ class Pubspec {
     }
   }
 
-  factory Pubspec.fromJson(Map json) => _$PubspecFromJson(json);
+  factory Pubspec.fromJson(Map json, {bool lenient = false}) {
+    lenient ??= false;
+
+    if (lenient) {
+      for (;;) {
```

Comment:
Should this be `while (json.isNotEmpty) {` ?

I could imagine a `CheckedFromJsonException` being thrown for a missing value – in which case this will loop forever, right?